commons/types/transaction.go

```diff
@@ -259,18 +265,27 @@ func TransactionPaging(txn *badger.Txn, startingHash string, page, pageSize int)
 			break
 		}
 	}
-	return transactions, nil //TODO: return error if empty?
+	if transactions[0] != nil {
+		data.Transactions = transactions
+		if startingHash != "" {
+			data.StartingHash = startingHash
+		} else {
+			data.StartingHash = transactions[0].Hash
```

Comment:
This logic is only true if page=1, which may not always be the case. What this should be is equal to the first iterator.Item() in the above for loop (which is sometimes skipped over).